web-client/public/js/update-app.js

```diff
@@ -124,6 +126,11 @@ const synchronizeHideAllWeights = () => {
     $(WEIGHTS_HIDE_CLASS).addClass("selected");
 };
 
+const synchronizeColorOptimalValues = function () {
+    $(COLOR_PREFERENCES_CLASS).toggleClass(ACTIVE_COLOR_OPTION);
+    $(COLOR_PREFERENCES_CLASS + ">span").toggleClass("glyphicon-ok invisible");
```

Comment:
I don't get this—`grnState` is not influencing this code at all. Instead, `toggleClass` is being used, which alternates between two states. That doesn't sound right. If `grnState.colorOptimization` changes in a way that is out-of-sync with toggling, then the states of the UI elements will be wrong.

Base your view code _directly_ on the status of `grnState`. `toggleClass` is reliant on the status quo—which isn't under the control of this code.